### PR TITLE
fix(datetime): confirm method now uses selected date

### PR DESCRIPTION
**Pull request by sean-perkins**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

When confirming the date using datetime's `confirm()` method, it will use the working parts (visible calendar month) instead of the active parts (selected calendar month). This results in incorrect date values when a user selects a date, navigates next/previous to a different calendar month and uses the `confirm()` API to close/dismiss the datetime overlay.

Issue URL: #24823

## What is the new behavior?

- The `confirm()` API uses the active parts when setting the current value.
- Navigating to next/previous months will not affect the date value when using `confirm()`, unless the user selects a new date inside that month.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by liamdebeasi — CHANGES_REQUESTED** [review]

**Review by amandaesmith3 — APPROVED** [review]
LGTM once Liam's comment is addressed 👍

File: core/src/components/datetime/test/confirm-date/index.html
Comment (sean-perkins) [reply core/src/components/datetime/test/confirm-date/index.html:15]: Is there a way to override the locale & value of the datetime for a specific test? I tried this:
```
await page.$eval('#modal ion-datetime', (el: any) => {
  el.locale = 'en-US';
  el.value = '2021-11-22';
});

await page.waitForChanges();
```

But it does not appear to override the initial values for the test. 

I'm open to other ways of testing this as well. I just need to open the datetime with a value, navigate to a next/prev month, use the `.confirm()` API and check the datetime's value.

File: core/src/components/datetime/test/basic/e2e.ts
Comment (sean-perkins) [inline core/src/components/datetime/test/basic/e2e.ts:127]: I need to fix this before merging.

**Pull request merged**
